vendor/github.com/docker/docker/pkg/parsers/parsers.go

```diff
@@ -52,6 +56,8 @@ func ParseUintListMaximum(val string, maximum int) (map[int]bool, error) {
 //	03,1-3      <- this is gonna get parsed as [1,2,3]
 //	3,2,1
 //	0-2,3,1
+//
+// Deprecated: ParseUintList was only used internally and will be removed in the next release.
 func ParseUintList(val string) (map[int]bool, error) {
```

Comment:
This is called from `pkg/sysinfo/sysinfo.go` , so further updates might be needed.

Reply:
… and they are.

Podman also calls this function; I guess c/common will need to carry it?

Reply:
I think so, maybe copy it to pkg/parse here.

Reply:
Done, PTAL.

Reply:
Note I wrote `pkg/parse` intentionally, not `pkg/parsers`.
While moving 1 to 1 like this makes sense I find it quite hard for callers to justify why there is a parse and parsers package here. (Not that most of the package structure makes sense here but we should not make it worse)

Reply:
> Note I wrote `pkg/parse` intentionally, not `pkg/parsers`.

I’m sorry, I missed that.

---

I _also_ missed something I should have checked sooner: we are already carrying a copy in c/storage; so, I have modified this PR to use that one, and now c/common has no direct dependency on Docker. I don’t think that helps all that much, Podman will still need to adjust for deprecations.

Reply:
Ack, yeah makes sense top use c/storage if we already carry the copy anyway there